docs/design/design_v1.8.md

```diff
@@ -0,0 +1,437 @@
+
+Please note that 
+
+1. The access to the `cluster-info` ConfigMap _is not_ rate-limited. This may or may not be a problem if you expose your master to the internet; worst-case scenario here is a DoS attack where an attacker uses all the in-flight requests the kube-apiserver can handle to serving the `cluster-info` ConfigMap.
+2. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+3. This phase can be invoked individually with the `kubeadm phase bootstrap-token node allow-auto-approve` command.
+
+### Install kube-proxy addon
+
+A ServiceAccount for `kube-proxy` is created in the `kube-system` namespace; then kube-proxy is deployed as a DaemonSet:
+
+- the credentials (`ca.crt` and `token`) to the master come from the ServiceAccount
+- the location of the master comes from a ConfigMap
+- the `kube-proxy` ServiceAccount is bound to the privileges in the `system:node-proxier` ClusterRole
+
+Please note that 
+
+1. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+2. This phase can be invoked individually with the `kubeadm phase addon kube-proxy`  command.
+
+### Install kube-dns addon
+
+A ServiceAccount for `kube-dns` is created in the `kube-system` namespace.
+
+Deploy the kube-dns Deployment and Service:
+
+- it's the upstream kube-dns deployment relatively unmodified
+- the `kube-dns` ServiceAccount is bound to the privileges in the `system:kube-dns` ClusterRole
+
+Please note that 
+
+1. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+2. This phase can be invoked individually with the `kubeadm phase addon kube-dns`  command.
+
+### (Optional) self-hosting
+
+This phase is performed only if `kubeadm init` is invoked with `—features-gates=self-hosting`
+
+The self hosting phase basically replaces static pods for control plane components with DaemonSets; this is achieved by executing following procedure for API Server, scheduler and controller manager static pods:
+
+- Load the Static Pod specification from disk 
+- Extract the PodSpec from that Static Pod specification
+- Mutate the PodSpec to be compatible with self-hosting, and more in detail:
+  - add node selector attribute targeting nodes with`node-role.kubernetes.io/master=""`  label, 
+  - add a toleration for `node-role.kubernetes.io/master:NoSchedule` taint,
+  - set `spec.DNSPolicy` to `ClusterFirstWithHostNet`
+- Build a new DaemonSet object for the self-hosted component in question. Use the above mentioned PodSpec
+- Create the DaemonSet resource in `kube-system` namespace. Wait until the Pods are running.
+- Remove the Static Pod manifest file. The kubelet will stop the original Static Pod-hosted component that was running
+
+Please note that:
+
+1. Self hosting is not yet resilient to node restarts; this can be fixed with external checkpointing; in 1.9 kubelet checkpointing will be available as well
```

Comment:
checkpointing for the control plane pods (not kubelet checkpointing, we don't want to call it generic yet really)